---
title: "Microbiome Education Poll Data Analysis - Supplementary Information"
output:
  pdf_document:
    toc: true
  html_document:
    toc: true
    number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, out.width = "60%")
```

# About

# Basic Poll Information

This poll was launched in May 2024 to better understand microbiome education, specifically focusing on the challenges and opportunities faced by instructors. The purpose of this poll was to gather insights and data that will support our efforts to promote equal access to microbiome training and resources.

The first poll response was collected May 20, 2024. A blank version of the poll can be viewed [here](https://drive.google.com/file/d/134MgEsKRB3GohHQTUAdQMruNI3yXmIwF/view?usp=drive_link). All code associated with this project can be found [on GitHub](https://github.com/fhdsl/microbiome-poll).

<!-- Setting up and cleaning the data in the chunk below -->

```{r}
library(googlesheets4) # gs4_auth read_sheet %>%
library(patchwork) # plot_layout plot_annotation
library(tidytext) # unnest_tokens
library(wordcloud) # wordcloud
library(SnowballC) # wordStem
library(scales) # percent_format
library(tidyverse) # Many :)

# Once set up, you can automate this process by passing your email
gs4_auth(email = TRUE)

# Read in the data from Google Sheets
sheet_url <-
  "https://docs.google.com/spreadsheets/d/1ECTBodmWRzbeCMSQROMJYh5H-2Tw6fFxdzFG7AShybg/edit?resourcekey=&gid=665284569#gid=665284569"
sheet_dat <- read_sheet(sheet_url)

# Clean data ---
# Make column names simpler
clean_names_dat <- sheet_dat %>%
  rename(
    familiar = 2,
    done_research = 3,
    offer_courses = 4,
    want_collab = 5,
    helps_students = 6,
    self_confidence = 7,
    dev_opportunities = 8,
    comp_access = 9,
    lab_access = 10,
    time_financial = 11,
    materials_access = 12,
    course_freedom = 13,
    student_interest = 14,
    barriers = 15,
    thoughts = 16,
    institution_type = 17,
    ammend = 18
  )

# Make cols true or false where appropriate
clean_TF_dat <- clean_names_dat %>%
  mutate(done_research = case_when(
    done_research == "No" ~ FALSE,
    done_research == "Yes" ~ TRUE,
    TRUE ~ NA
  )) %>%
  mutate(offer_courses = case_when(
    offer_courses == "No" ~ FALSE,
    offer_courses == "Yes" ~ TRUE,
    TRUE ~ NA
  ))

# Function to clean up institution types and one-hot encode
make_inst_categories <- function(data, newcol, string) {
  return(data %>%
           mutate({{newcol}} := case_when(
             str_detect(institution_type, string) ~ TRUE, TRUE ~ FALSE
           )))
}

# Clean institution types
clean_types_dat <-
  clean_TF_dat %>%
  make_inst_categories("r1", "Doctoral Universities – Very high research activity") %>%
  make_inst_categories("r2", "Doctoral Universities – High research activity") %>%
  make_inst_categories("masters", "Master’s College or University") %>%
  make_inst_categories("pui", "Primarily Undergraduate Institution") %>%
  make_inst_categories("cc", "Community College") %>%
  make_inst_categories("hbcu", "Historically Black College or University") %>%
  make_inst_categories("hsi", "Hispanic Serving Institution") %>%
  make_inst_categories("tcu", "Tribal College or University") %>%
  make_inst_categories("nasnti", "Native American-Serving, Nontribal Institutions") %>%
  make_inst_categories("aanapisi",
                       "Asian American and Native American Pacific Islander-Serving Institution") %>%
  make_inst_categories("other_uri", "Other under-resourced institution") %>%
  make_inst_categories("none_type", "None of these")

# Make a small manual adjustment (someone emailed about a response correction)
clean_types_dat <-
  clean_types_dat %>%
  mutate(hsi = case_when(
    ammend == "Emailed to say they meant to also select HSI" ~ TRUE,
    TRUE ~ hsi
  ))

# Clean up Carnegie Classification. An institution can't really be more than one.
clean_types_dat <-
  clean_types_dat %>%
  left_join(clean_types_dat %>%
              filter(r1) %>%
              mutate(r1_r = TRUE)) %>%
  left_join(clean_types_dat %>%
              filter(r2 & !r1) %>%
              mutate(r2_r = TRUE)) %>%
  left_join(clean_types_dat %>%
              filter(masters & !r1 & !r2) %>%
              mutate(m_r = TRUE)) %>%
  left_join(clean_types_dat %>%
              filter(pui & !r1 & !r2 & !masters) %>%
              mutate(pui_r = TRUE)) %>%
  left_join(clean_types_dat %>%
              filter(cc & !r1 & !r2 & !masters & !pui) %>%
              mutate(cc_r = TRUE)) %>%
  mutate(across(ends_with("_r"), ~ ifelse(is.na(.x), F, T))) %>%
  mutate(
    r1 = r1_r,
    r2 = r2_r,
    masters = m_r,
    pui = pui_r,
    cc = cc_r
  ) %>%
  select(!ends_with("_r")) %>%
  mutate(unknown_classification = case_when(!r1 &
                                              !r2 &
                                              !masters & !pui & !cc ~ TRUE, TRUE ~ FALSE))

# Make a column for "research" institutions
clean_types_dat <-
  clean_types_dat %>%
  mutate(research_type = case_when(r1 | r2 | masters ~ TRUE, TRUE ~ FALSE))

# Make a column for MSI institutions
# QUESTION: Include Community Colleges in MSIs? See below
clean_types_dat <-
  clean_types_dat %>%
  mutate(msi_type = case_when(hbcu |
                                hsi | tcu | nasnti | aanapisi | other_uri ~ TRUE, TRUE ~ FALSE))
```

```{r}
get_se <- function(x){sd(x, na.rm = T) / sqrt(length(na.omit(x)))}
```

<!-- Gut check below -->

```{r eval=FALSE}
# Is there a lot of overlap for CCs and MSIs?
clean_types_dat %>% filter(cc & !msi_type) # ~7 responses
```

## Response Rate

```{r}
sample_size <- nrow(clean_types_dat)
total_research_pool_n <- 291
response_rate <- round(sample_size / total_research_pool_n * 100, 2)

# Create breakdown tables of institution types
rsch_breakdown <-
  clean_types_dat %>% select(r1:cc, unknown_classification) %>% rename(
    `R1 institution` = r1,
    `R2 institution` = r2,
    `Master's degree granting institution` = masters,
    `Primarily undergraduate institution` = pui,
    `Community college` = cc,
    `Unknown` = unknown_classification
  ) %>%
  pivot_longer(everything()) %>%
  filter(value) %>%
  group_by(name) %>%
  count() %>%
  arrange(desc(n)) %>%
  mutate(percent = paste0(round(n / sample_size * 100), "%"))

msi_breakdown <-
  clean_types_dat %>%
  select(hbcu:other_uri) %>%
  rename(
    `Hispanic Serving Institution` = hsi,
    `Historically Black College or University` = hbcu,
    `Other under-resourced institution` = other_uri,
    `Asian American and Native American Pacific Islander Serving Institution` = aanapisi,
    `Tribal College or University` = tcu
  ) %>%
  pivot_longer(everything()) %>%
  filter(value) %>%
  group_by(name) %>%
  count() %>%
  arrange(desc(n)) %>%
  mutate(percent = paste0(round(n / sample_size * 100), "%"))

```

The poll had `r sample_size` respondents, for a response rate of `r response_rate`%.

## Institution Types

```{r}
kableExtra::kable(rsch_breakdown, align = "lrr") %>%
  kableExtra::kable_styling(full_width = F, position = "left")
kableExtra::kable(msi_breakdown, align = "lrr") %>%
  kableExtra::kable_styling(full_width = F, position = "left")
```

# Responses: General Poll Questions

```{r eval=FALSE}
# Function to create side by side column plots
do_col_plot <- function(y_attribute) {
  # Get subset broken down by research_type
  gg_rsch <-
    clean_types_dat %>%
    group_by(research_type) %>%
    summarize(mean = mean(familiar, na.rm = T),
              sd = get_se(familiar)) %>%
    ggplot(aes(x = research_type, y = mean)) +
    geom_col(fill = c("#21918c", "#3b528b")) +
    scale_x_discrete(
      breaks = c("FALSE", "TRUE"),
      labels = c(
        "Primarily\nUndergraduate\n Institution",
        "Graduate\nResearch\n Institution"
      )
    )
  
  # Get subset broken down by msi_type
  gg_msi <-
    clean_types_dat %>%
    group_by(msi_type) %>%
    summarize(mean = mean(!!sym(y_attribute), na.rm = T),
              sd = get_se(!!sym(y_attribute))) %>%
    ggplot(aes(x = msi_type, y = mean)) +
    geom_col(fill = c("#fde725", "#5ec962")) +
    scale_x_discrete(breaks = c("FALSE", "TRUE"),
                     labels = c("Non-MSI", "MSI"))
  
  return(
    require(patchwork)
    gg_rsch + gg_msi &
      theme(legend.position = "none") &
      geom_errorbar(aes(
        ymin = mean - sd,
        ymax = mean + sd,
        width = 0.4
      )) &
      labs(x = NULL) &
      theme_minimal()
  )
}

# Function to create side by side column plots, with a yes/no binary response
do_yn_plot <- function(y_attribute, legend_title) {
  # Get subset broken down by research_type
  gg_rsch <-
    clean_types_dat %>%
    group_by(research_type, !!sym(y_attribute)) %>%
    summarize(n = n()) %>%
    group_by(research_type) %>% 
    mutate(percent = n / sum(n) * 100) %>% 
    ggplot(aes(x = research_type, y = percent, fill = !!sym(y_attribute))) +
    geom_col() +
    scale_fill_manual(values = c("#3b528b", "#5ec962", "gray70")) +
    scale_x_discrete(
      breaks = c("FALSE", "TRUE"),
      labels = c(
        "Primarily\nUndergraduate\n Institution",
        "Graduate\nResearch\n Institution"
      )
    ) +
    labs(y = "% Respondents") +
    theme_minimal() +
    theme(legend.position = "none")
  
  # Get subset broken down by msi_type
  gg_msi <-
    clean_types_dat %>%
    group_by(msi_type, !!sym(y_attribute)) %>%
    summarize(n = n()) %>%
    group_by(msi_type) %>% 
    mutate(percent = n / sum(n) * 100) %>% 
    ggplot(aes(x = msi_type, y = percent, fill = !!sym(y_attribute))) +
    geom_col() +
    scale_fill_manual(values = c("#3b528b", "#5ec962", "gray70")) +
    scale_x_discrete(breaks = c("FALSE", "TRUE"),
                     labels = c("Non-MSI", "MSI")) +
    labs(y = "% Respondents") +
    theme_minimal() +
    theme(legend.position = "bottom",
          legend.justification.bottom = "right")
  
  return(
    require(patchwork)
    gg_rsch + gg_msi &
      labs(x = NULL,
           fill = legend_title)
  )
}

# Equivalent of do_col_plot but don't break it down by research type or msi type
do_col_plot_no_grouping <- function(y_attribute) {
  gg_all <-
    clean_types_dat %>%
    summarize(mean = mean(!!sym(y_attribute), na.rm = T),
              sd = get_se(!!sym(y_attribute))) %>%
    mutate(x = "All respondents") %>%
    ggplot(aes(x = x, y = mean)) +
    geom_col(fill = c("grey30"))
  
  return(
    gg_all &
      theme(legend.position = "none") &
      geom_errorbar(aes(
        ymin = mean - sd,
        ymax = mean + sd,
        width = 0.4
      )) &
      labs(x = NULL) &
      theme_minimal()
  )
}

# Equivalent of do_yn_plot but don't break it down by research type or msi type
do_yn_plot_no_grouping <- function(y_attribute, legend_title) {
  gg_all <-
    clean_types_dat %>%
    group_by(!!sym(y_attribute)) %>%
    summarize(n = n()) %>%
    mutate(percent = n / sum(n) * 100) %>%
    mutate(x = "All respondents") %>%
    ggplot(aes(
      x = x,
      y = percent,
      fill = !!sym(y_attribute)
    )) +
    geom_col() +
    scale_fill_manual(values = c("#3b528b", "#5ec962", "gray70")) +
    labs(y = "% Respondents") +
    theme_minimal() +
    theme(legend.position = "bottom",
          legend.justification.bottom = "right")
  
  return(gg_all &
           labs(x = NULL, fill = legend_title))
}
```

<br>

## Are you familiar with microbiome research and its potential applications in public health, medicine, the environment, and industry?

```{r eval=FALSE}
p1 <- do_col_plot("familiar") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
p2 <- do_col_plot_no_grouping("familiar")
p1 + p2 
```

```{r eval=FALSE}
# Response on average:
clean_types_dat %>%
  summarize(mean = mean(familiar, na.rm = T))
```

<br>

## Have you conducted microbiome research, previously or currently?

```{r eval=FALSE}
do_yn_plot("done_research", legend_title = "")
```

```{r eval=FALSE}
# What's the breakdown?
clean_types_dat %>%
  group_by(done_research) %>%
  summarize(n = n()) %>%
  mutate(freq = n / sum(n))

clean_types_dat %>%
  group_by(research_type, done_research) %>%
  summarize(n = n()) %>%
  mutate(freq = n / sum(n))

clean_types_dat %>%
  group_by(msi_type, done_research) %>%
  summarize(n = n()) %>%
  mutate(freq = n / sum(n))
```

<br>

## Do you, or others at your institution, offer specific courses or educational programs for students related to microbiomes?

```{r eval=FALSE}
do_yn_plot("offer_courses", legend_title = "")
```

```{r eval=FALSE}
# What's the breakdown?
clean_types_dat %>%
  group_by(offer_courses) %>%
  summarize(n = n()) %>%
  mutate(freq = n / sum(n))

clean_types_dat %>%
  group_by(research_type, offer_courses) %>%
  summarize(n = n()) %>%
  mutate(freq = n / sum(n))
```

<br>

## Would you be interested in collaborating with other institutions, organizations, or faculty to enhance microbiome education or research opportunities at your institution?

```{r eval=FALSE}
do_col_plot("want_collab") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish) 
```

```{r eval=FALSE}
#Response on average:
clean_types_dat %>%
  summarize(mean = mean(want_collab, na.rm = T))
```

<br>

## Do you believe that incorporating microbiome topics into the curriculum can provide students with interdisciplinary skills to address real-world problems?

```{r eval=FALSE}
do_col_plot("helps_students") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish) 
```

```{r eval=FALSE}
# Response on average:
clean_types_dat %>%
  summarize(mean = mean(helps_students, na.rm = T))

clean_types_dat %>%
  group_by(helps_students) %>%
  count()
```

<br>

# Responses: Institutional Barriers

## Institutional Barriers (separated)

The following are visualizations of institution types by each barrier. Note that research institution status and MSI status are not mutually exclusive. For example, a research institution could be either non-MSI or MSI.

_When implementing or considering implementing a module or course, please indicate the extent of each of the following challenges, where 1 is "not a challenge" and 5 is "a strong challenge"._

<br>

### Confidence and/or expertise for teaching the material

```{r eval=FALSE}
do_col_plot("self_confidence") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Opportunities for development and/or access to a community of practice

```{r eval=FALSE}
do_col_plot("dev_opportunities") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Access to computers and/or computational time

```{r eval=FALSE}
do_col_plot("comp_access") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Access to laboratory supplies to incorporate wet lab activities

```{r eval=FALSE}
do_col_plot("lab_access") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Time and/or financial support needed to design and/or implement courses

```{r eval=FALSE}
do_col_plot("time_financial") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Access to existing course materials

```{r eval=FALSE}
do_col_plot("materials_access") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Freedom and/or flexibility to implement a course or module of your choosing

```{r eval=FALSE}
do_col_plot("course_freedom") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

### Student interest and/or engagement

```{r eval=FALSE}
do_col_plot("student_interest") &
  scale_y_continuous(limits = c(0, 5), oob = scales::oob_squish)
```

<br>

## Institutional Barriers (aggregated)

_When implementing or considering implementing a module or course, please indicate the extent of each of the following challenges, where 1 is "not a challenge" and 5 is "a strong challenge"._

```{r}
# Make a big plot of all the barriers
do_challenge_plot <- function() {
  rsch_dat <-
    clean_types_dat %>%
    select(7:14, research_type) %>%
    pivot_longer(!research_type) %>%
    group_by(research_type, name) %>%
    summarize(mean = mean(value), sd = get_se(value)) %>%
    mutate(
      type = case_when(
        !research_type ~ "Primarily Undergraduate Institution",
        research_type ~ "Graduate Research Institution"
      )
    )
  msi_dat <-
    clean_types_dat %>%
    select(7:14, msi_type) %>%
    pivot_longer(!msi_type) %>%
    group_by(msi_type, name) %>%
    summarize(mean = mean(value), sd = get_se(value)) %>%
    mutate(type = case_when(!msi_type ~ "Non-MSI", msi_type ~ "MSI"))
  
  gg_out <-
    bind_rows(rsch_dat, msi_dat) %>%
    mutate(type = factor(
      type,
      levels = c(
        "Primarily Undergraduate Institution",
        "Graduate Research Institution",
        "Non-MSI",
        "MSI"
      )
    )) %>%
    mutate(
      name = case_when(
        name == "comp_access" ~ "Access to computers and/or\ncomputational time",
        name == "course_freedom" ~ "Freedom and/or flexibility to implement a\ncourse or module of your choosing",
        name == "dev_opportunities" ~ "Opportunities for development and/or\naccess to a community of practice",
        name == "lab_access" ~ "Access to laboratory supplies to\nincorporate wet lab activities",
        name == "materials_access" ~ "Access to existing course materials",
        name == "self_confidence" ~ "Confidence and/or expertise for\nteaching the material",
        name == "student_interest" ~ "Student interest and/or engagement",
        name == "time_financial" ~ "Time and/or financial support needed to\ndesign and/or implement courses"
      )
    ) %>%
    ggplot(aes(
      x = forcats::fct_reorder(name, mean, .desc = T),
      y = mean,
      fill = type,
      group = type
    )) +
    geom_col(position = position_dodge(width = 0.9)) +
    theme(axis.text.x = element_text(angle = 30)) +
    geom_errorbar(position = position_dodge(width = 0.9),
                  aes(
                    ymin = mean - sd,
                    ymax = mean + sd,
                    width = 0.4
                  )) +
    labs(x = NULL) +
    scale_fill_manual(values = c("#21918c", "#3b528b", "#fde725", "#5ec962")) +
    theme_minimal() +
    theme(
      legend.title = element_blank(),
      axis.text.x = element_text(angle = 50, hjust = 1),
      plot.margin = unit(c(1, 1, 1, 2), "cm"),
      text = element_text(size = 7)
    )
  return(gg_out)
}
```

```{r out.width="100%"}
# Make the aggregated plot
do_challenge_plot()
```

# Statistics

## Would you be interested in collaborating with other institutions, organizations, or faculty to enhance microbiome education or research opportunities at your institution?

Based on the outcome below, MSI faculty rank interest in collaborating higher.

```{r}
t_msi <- clean_types_dat %>% filter(msi_type == TRUE) %>% pull(want_collab)
t_nonmsi <- clean_types_dat %>% filter(msi_type == FALSE) %>% pull(want_collab)
t.test(t_msi, t_nonmsi, alternative = "two.sided", var.equal = FALSE)
wilcox.test(t_msi, t_nonmsi, alternative = "two.sided")
```

## Logistic Regression of Barriers

The following is a poisson regression model looking at how research type ("Primarily Undergraduate Institution" vs. "Graduate Research Institution") and is/is not an MSI affect the perception on different barriers to implementing microbiome work. 

Based on the outcome below, there isn't evidence that research type / MSI affects these perceived barriers. However, "Time and/or financial support needed to design and/or implement courses" is the biggest barrier for all institutions.

```{r}
stats_dat <-
  clean_types_dat %>%
  select(7:14, research_type, msi_type) %>%
  pivot_longer(!c(research_type, msi_type))

summary(glm(
  data = stats_dat,
  value ~ name + research_type + msi_type,
  family = poisson(link = "log")
))
```


# Free Text Thematic Analysis

We used a qualitative research approach called thematic analysis ([Braun and Clarke 2006](https://www.tandfonline.com/doi/abs/10.1191/1478088706QP063OA)) to better understand free text responses. This approach has previously been used to better understand instructor perspectives. For example, it has been used to understand instructor perspectives on the use of LLMs in the classroom ([link](https://arxiv.org/abs/2309.10694)), blended learning ([link](https://onlinelibrary.wiley.com/doi/abs/10.1111/jcal.12578?casa_token=6vdUGgxugZoAAAAA:Wc2cQx7XchX3NTU4fnXGKJjLfIVnFSJTm0UyG8dpD1pXnuICvoK5Bg3GjKdmLHelqp4HIXVws5KHbA)), professional development opportunities ([link](https://link.springer.com/article/10.1007/s12528-018-9194-0)), and threshold concepts ([link](https://doi.org/10.1145/2700513)). It has been used extensively to better understand STEM education ([link](https://link.springer.com/article/10.1186/s43031-024-00106-8), [link](https://link.springer.com/article/10.1186/s40594-017-0062-7), [link](https://link.springer.com/article/10.1007/s41979-023-00103-y)).

```{r}
words_dat <-
  clean_types_dat %>%
  mutate(response_number = row_number())
```

## Barrier Thematic Analysis

_"Please provide any additional thoughts on barriers you have encountered with microbiome education."_

**Overall Barrier Themes**

There might be some disagreement about what constitutes microbiome education at different institution types. For example, a CURE at a non-research institution might leverage existing data, while a research institution might feel that expensive original data acquisition is required.

* Sequencing was mentioned repeatedly for non-MSIs but not at MSIs
* Costs were mentioned at research institutions but not at undergraduate institutions.
* CUREs were mentioned 6 times at non-research MSI institutions, but not at all at research institutions / non-MSI institutions.

**Themes for barriers at MSIs**

1. Curriculum integration and time constraints

* "We are a two-year college offering lower-division undergraduate coursework, so it can be challenging to incorporate advanced topics and areas of active research into the curriculum."
* "Integrating the topic into existing learning outcomes of a course. For our institution, we won't be able to offer a specific course on microbiome, but I can see several courses into which microbiome lessons can be integrated."
* "Primarily the time to research it and the availability of teaching resources"
* "may need some training and lab equipment to conduct such lab module"
* "Access to researchers with correct expertise and TIME to collaborate"
* "Computational and statistical expertise to support microbiomes studies is lacking at our institution. We have tried reaching out to computer science and statistics faculty but there has been very limited interest and/or experience in this specific application to microbiomes data."

2. Student engagement and background

* "Student understanding of technologies; lack of resources to incorporate activities in student labs. Lack of student preparedness for lower-level microbiology course."
* "Students are interested when microbiome research is discussed but often lack knowledge"
* "Bioinformatic flows are difficult for students to learn. Thus, they become frustrated. Many of my students state they prefer the wet-bench portion of metagenomics."
* "It is hard to get students excited or engaged in anything. However, students were more interested when they were presented with examples, articles, social media clickbait of how it relates to them and their health."

**Themes for barriers at non-MSIs**

1. Cost barriers for new sequencing

* "Money and time and training are the big three"
* "Lack of funding for sequencing"
* "Lack of funds for expensive reagents and whole genome sequencing."
* "Cost per sample and turnaround time for receiving data if using an external vendor could be barriers for having students do the DNA sequencing for microbial communities in class."
* "Cost and resources are a really big factors. I can teach all day about the theory behind microbiome concepts and techniques, but doing and practicing the techniques can be quite difficult."
* "Cost for supplies and sequencing if generating data and updating computers for analysis are both challenges at my institution."
* "Paying for sequencing and knowing how to do the bioinformatics"
* "They are listed above - finances and equipment"
* "It is too expensive" 

2. Instructor skills and confidence

* "My own ignorance and weaknesss is a big roadblock."
* "Some challenges are library prep and makes sense out [of] .. output."
* "I would love hands on training and a network. It would give me to confidence I need to create a whole class around it rather than just a module or assignment."
* "As an instructor, I lack some of the statistical background to explain in detail when community compositions are different. I would love to have (and contribute to) resources available for this."
* "Paying for sequencing and knowing how to do the bioinformatics"

**Themes for barriers at non-research institutions**

These themes overlap with MSIs (above) including lack of student background, flexibility to implement microbiome content, and time/expertise constraints.

```{r eval = FALSE}
words_dat %>%
  rename(text_col = barriers, case_col = research_type) %>%
  select(response_number, text_col, case_col) %>%
  unnest_tokens(word, text_col) %>%
  anti_join(stop_words[stop_words$lexicon == "SMART", ], by = "word") %>%
  mutate(stem = wordStem(word)) %>%
  filter(!(word %in% c("na"))) %>%
  drop_na(word) %>%
  count(case_col, word) %>%
  pivot_wider(names_from = case_col, values_from = n) %>%
  filter(is.na(`TRUE`)) %>%
  arrange(desc(`FALSE`))
```

**Themes for barriers at research institutions**

Costs and time, either from the instructor themselves or lack of personnel or collaborators.

## Other Thoughts Thematic Analysis

_"Please provide any additional thoughts on your experience with microbiome education."_

**Overall Thoughts Themes**

There were fewer divisions among types of institutions. In general, respondents felt a lot of enthusiasm for the topic and for collaboration. There is some concern around how to implement microbiome content in the classroom, as well as expertise and time to make that happen.

**Other Thoughts Themes at MSIs**

1. Enthusiasm for the topic

* "Very important and ties into many content areas, there are lots of misconceptions on the part of the students."
* "I do think this is an outstanding topic with tons of potential and it should be part of the articulation agreements."
* "Microbiome studies are a particularly good subject for learning the interdisciplinary nature of biology and integrating molecular biology, microbiology, ecology and environmental sciences.  It is an area rich in unanswered questions so it is perfect for CURE learning and teaching."
* "It is a huge passion of mine that I always include in my courses but never as a “module” so I would be very excited to see collaboration"
* "I do discuss in my courses, and would like to implement it more"                                                               

2. Enthusiasm for collaboration

* "My colleague (a chemist/environmental scientist who studies heavy-metal pollution) and I have been developing a collaboration to study microbes living in mine water drainage remediation systems. We are interested in incorporating this project into our courses."
* "I am excited to collaborate with researchers and educators at other institutions to receive training/support that will advance microbiome studies at our campus"
* "The desire is there, finding Microbiome experts has been challenging"                                                          

**Other Thoughts Themes at non-MSIs**

1. Enthusiasm for the topic and partnering with others

* "My Introduction to Microbiology students are very interested in the complexities and presence of microbial communities!  This is very much an interesting and important area of research and I already have plans to start developing a class focused in this direction."
* "I just briefly do some case studies and discussions about microbiome work, but I would love to do more"
* "I have been working on a collaborative project with my colleague in environmental to study microbial communities in acid mine drainage remediation systems, which is a project we are interested in building on."
* "This is a topic that is very important for me to teach and I incorporate it into my nutrition and biodiversity classes. Ideas for ways to teach it are always welcome." 
* "REMnet and Microbiomes for all has been a big help."

2. More planning is needed in how to integrate microbiome topics appropriately

* "It’s very powerful and interesting, but I believe it is better suited to higher level courses than first -year general biology"
* "I'm still unsure how to design enduring learning outcomes at the undergrad level regarding microbiomes." 
* "It is tricky to know how to make space in the curriculum"

3. Lack of time to learn or implement

* "I'd love to jump back into iPython/Jupyter notebooks but that requires time to develop I don't really have in my heavy teaching load position."
* "time to teach something like R and computing power/software accessibility have prevented me from trying to implement this level of analysis in my classes."
* "If instructors weren’t trained during grad or post-doc it is harder to find the right place for training"

```{r eval = FALSE}
words_dat %>%
  rename(text_col = thoughts, case_col = research_type) %>%
  select(response_number, text_col, case_col) %>%
  unnest_tokens(word, text_col) %>%
  anti_join(stop_words[stop_words$lexicon == "SMART", ], by = "word") %>%
  mutate(stem = wordStem(word)) %>%
  filter(!(word %in% c("na"))) %>%
  drop_na(word) %>%
  count(case_col, word) %>%
  pivot_wider(names_from = case_col, values_from = n) %>%
  filter(is.na(`TRUE`)) %>%
  arrange(desc(`FALSE`))
```

## Wordclouds

```{r}
words_barriers <- words_dat %>%
  select(response_number, barriers) %>%
  unnest_tokens(word, barriers)
words_thoughts <- words_dat %>%
  select(response_number, thoughts) %>%
  unnest_tokens(word, thoughts)
```

_"Please provide any additional thoughts on barriers you have encountered with microbiome education."_

```{r}
words_barriers %>%
  anti_join(stop_words) %>%
  count(word) %>%
  with(wordcloud(word, n, max.words = 100))
```

_"Please provide any additional thoughts on your experience with microbiome education."_

```{r}
words_thoughts %>%
  anti_join(stop_words) %>%
  count(word) %>%
  with(wordcloud(word, n, max.words = 100))
```

## Word Frequencies: MSIs versus Non-MSIs

```{r}
make_text_freq <- function(in_df) {
  text_df <-
    in_df %>%
    select(response_number, text_col, case_col) %>%
    unnest_tokens(word, text_col) %>%
    anti_join(stop_words[stop_words$lexicon == "SMART", ], by = "word") %>%
    mutate(stem = wordStem(word)) %>%
    filter(!(word %in% c("na"))) %>%
    drop_na(word) %>%
    count(case_col, word) %>%
    group_by(case_col) %>%
    mutate(proportion = n / sum(n)) %>%
    select(!n) %>%
    pivot_wider(names_from = case_col, values_from = proportion)
  
  text_df <- text_df[!grepl("\\d", text_df$word), ] # Remove numbers
  
  # expect a warning about rows with missing values being removed
  gg <-
    ggplot(text_df, aes(
      x = `TRUE`,
      y = `FALSE`,
      color = abs(`FALSE` - `TRUE`)
    )) +
    geom_abline(color = "gray20", lty = 2) +
    geom_jitter(
      alpha = 0.3,
      size = 2.5,
      width = 0.05,
      height = 0.05
    ) +
    theme_minimal() +
    geom_text(aes(label = word), check_overlap = TRUE) +
    scale_x_log10(labels = percent_format()) +
    scale_y_log10(labels = percent_format()) +
    scale_color_gradient(low = "gray20", high = "#5ec962") +
    theme(legend.position = "none")
  
  return(gg)
}
```

Below shows the differences between frequency of certain words. Words farther away from the midpoint are more commonly found in that institution type. 

The word pool is the following question.

_"Please provide any additional thoughts on barriers you have encountered with microbiome education."_

```{r}
words_dat %>%
  rename(text_col = barriers, case_col = msi_type) %>% make_text_freq() +
  labs(y = "Not an MSI", x = "MSI")
```

The word pool is the following question.

_"Please provide any additional thoughts on your experience with microbiome education."_

```{r}
words_dat %>%
  rename(text_col = thoughts, case_col = msi_type) %>% make_text_freq() +
  labs(y = "Not an MSI", x = "MSI")
```

The word pool is both questions above.

```{r}
words_dat %>%
  mutate(total_text = paste0(barriers, ". ", thoughts)) %>%
  rename(text_col = total_text, case_col = msi_type) %>% make_text_freq() +
  labs(y = "Not an MSI", x = "MSI")
```

## Word Frequencies: Graduate institutions vs. undergraduate institutions

Below shows the differences between frequency of certain words. Words farther away from the midpoint are more commonly found in that institution type. 

The word pool is the following question.

_"Please provide any additional thoughts on barriers you have encountered with microbiome education."_

```{r}
words_dat %>%
  rename(text_col = barriers, case_col = research_type) %>% make_text_freq() +
  labs(y = "Undergraduate institution", x = "Graduate institution")
```

The word pool is the following question.

_"Please provide any additional thoughts on your experience with microbiome education."_

```{r}
words_dat %>%
  rename(text_col = thoughts, case_col = research_type) %>% make_text_freq() +
  labs(y = "Undergraduate institution", x = "Graduate institution")
```

The word pool is both questions above.

```{r}
words_dat %>%
  mutate(total_text = paste0(barriers, ". ", thoughts)) %>%
  rename(text_col = total_text, case_col = research_type) %>% make_text_freq() +
  labs(y = "Undergraduate institution", x = "Graduate institution")
```

# Figures for the publication

Please see the repository: https://github.com/fhdsl/microbiome-poll for rendered figures.

```{r eval=FALSE}
### FIGURE 2 ###

font_size_ <- 10

#####
p1 <-
  clean_types_dat %>%
  summarize(mean = mean(familiar, na.rm = T),
            sd = get_se(familiar)) %>%
  mutate(y = "Are you familiar with microbiome research and its potential applications\nin public health, medicine, the environment, and industry?") %>%
  rbind(
    clean_types_dat %>%
      summarize(
        mean = mean(helps_students, na.rm = T),
        sd = get_se(helps_students)
      ) %>%
      mutate(y = "Do you believe that incorporating microbiome topics into the curriculum can\nprovide students with interdisciplinary skills to address real-world problems?")
  ) %>%
  ggplot(aes(x = mean, y = y)) +
  geom_col(fill = c("grey30")) +
  geom_errorbar(aes(
    xmin = mean - sd,
    xmax = mean + sd,
    width = 0.4
  )) +
  labs(x = NULL, y = NULL) +
  theme_minimal() +
  theme(legend.position = "none",
        axis.text = element_text(size = font_size_))
#####
p2 <-
  clean_types_dat %>%
  group_by(research_type, offer_courses) %>%
  summarize(n = n()) %>%
  group_by(research_type) %>%
  mutate(
    percent = n / sum(n) * 100,
    facet = "Research type",
    type = case_when(
      research_type ~ "Graduate Research Institution",
      !research_type ~ "Primarily Undergraduate Institution"
    )
  ) %>%
  rbind(
    clean_types_dat %>%
      group_by(msi_type, offer_courses) %>%
      summarize(n = n()) %>%
      group_by(msi_type) %>%
      mutate(
        percent = n / sum(n) * 100,
        facet = "MSI type",
        type = case_when(msi_type ~ "MSI", !msi_type ~ "non-MSI")
      )
  ) %>%
  rbind(
    clean_types_dat %>%
      group_by(offer_courses) %>%
      summarize(n = n()) %>%
      mutate(
        percent = n / sum(n) * 100,
        facet = " ",
        type = "Do you, or others at your institution, offer specific\ncourses or educational programs for students related\nto microbiomes? (All Respondents)"
      )
  ) %>%
  ggplot(aes(y = type, x = percent, fill = offer_courses)) +
  geom_col() +
  scale_fill_manual(values = c("#3b528b", "#5ec962", "gray70")) +
  facet_wrap( ~ facet,
              scales = "free_y",
              nrow = 3,
              strip.position = "right") +
  labs(x = "% Respondents") +
  theme_minimal() +
  theme(
    legend.position = "none",
    strip.text.y.right = element_text(angle = 0),
    plot.title = element_text(size = font_size_, hjust = 0),
    axis.text = element_text(size = font_size_)
  ) +
  labs(y = NULL, fill = "")

p3 <-
  clean_types_dat %>%
  group_by(research_type, done_research) %>%
  summarize(n = n()) %>%
  group_by(research_type) %>%
  mutate(
    percent = n / sum(n) * 100,
    facet = "Research type",
    type = case_when(
      research_type ~ "Graduate Research Institution",
      !research_type ~ "Primarily Undergraduate Institution"
    )
  ) %>%
  rbind(
    clean_types_dat %>%
      group_by(msi_type, done_research) %>%
      summarize(n = n()) %>%
      group_by(msi_type) %>%
      mutate(
        percent = n / sum(n) * 100,
        facet = "MSI type",
        type = case_when(msi_type ~ "MSI", !msi_type ~ "non-MSI")
      )
  ) %>%
  bind_rows(
    clean_types_dat %>%
      group_by(done_research) %>%
      summarize(n = n()) %>%
      mutate(
        percent = n / sum(n) * 100,
        facet = " ",
        type = "Have you conducted microbiome research,\npreviously or currently? (All Respondents)"
      )
  ) %>%
  ggplot(aes(y = type, x = percent, fill = done_research)) +
  geom_col() +
  scale_fill_manual(values = c("#3b528b", "#5ec962"),
                    labels = c("No", "Yes")) +
  facet_wrap( ~ facet,
              scales = "free_y",
              nrow = 3,
              strip.position = "right") +
  labs(x = "% Respondents") +
  theme_minimal() +
  theme(
    legend.position = "bottom",
    strip.text.y.right = element_text(angle = 0),
    plot.title = element_text(size = font_size_, hjust = 0),
    axis.text = element_text(size = font_size_)
  ) +
  labs(y = NULL, fill = "")

p4 <- clean_types_dat %>%
  group_by(research_type) %>%
  summarize(mean = mean(want_collab, na.rm = T),
            sd = get_se(want_collab)) %>%
  mutate(
    facet = "Research type",
    type = case_when(
      research_type ~ "Graduate Research Institution",
      !research_type ~ "Primarily Undergraduate Institution"
    )
  ) %>%
  bind_rows(
    clean_types_dat %>%
      group_by(msi_type) %>%
      summarize(
        mean = mean(want_collab, na.rm = T),
        sd = get_se(want_collab)
      ) %>%
      mutate(
        facet = "MSI type",
        type = case_when(msi_type ~ "MSI", !msi_type ~ "non-MSI")
      )
  ) %>%
  bind_rows(
    clean_types_dat %>%
      summarize(
        mean = mean(want_collab, na.rm = T),
        sd = get_se(want_collab)
      ) %>%
      mutate(facet = " ", type =  "Would you be interested in collaborating with other institutions,\norganizations, or faculty to enhance microbiome education or research\nopportunities at your institution? (All Respondents)")
  ) %>%
  ggplot(aes(x = mean, y = type)) +
  geom_col(fill = c("grey30")) +
  geom_errorbar(aes(
    xmin = mean - sd,
    xmax = mean + sd,
    width = 0.4
  )) +
  facet_wrap(~ facet,
             scales = "free_y",
             nrow = 3,
             strip.position = "right") +
  labs(x = NULL, y = NULL) +
  theme_minimal() +
  theme(
    legend.position = "none",
    axis.text = element_text(size = font_size_),
    strip.text.y.right = element_text(angle = 0)
  )
#####

p1 + p2 + p3 + p4 + plot_layout(ncol = 1, heights = c(1.5, 2, 2, 2)) + plot_annotation(tag_levels = 'A')

ggsave(
  "figures/fig2.png",
  height = 9,
  width = 8,
  units = "in"
)
```

```{r eval=FALSE}
### FIGURE 3 ###


rsch_dat <-
  clean_types_dat %>%
  select(7:14, research_type) %>%
  pivot_longer(!research_type) %>%
  group_by(research_type, name) %>%
  summarize(mean = mean(value), sd = get_se(value)) %>%
  mutate(
    type = case_when(
      !research_type ~ "Primarily Undergraduate Institution",
      research_type ~ "Graduate Research Institution"
    )
  )
msi_dat <-
  clean_types_dat %>%
  select(7:14, msi_type) %>%
  pivot_longer(!msi_type) %>%
  group_by(msi_type, name) %>%
  summarize(mean = mean(value), sd = get_se(value)) %>%
  mutate(type = case_when(!msi_type ~ "Non-MSI", msi_type ~ "MSI"))

gg_out <-
  bind_rows(rsch_dat, msi_dat) %>%
  mutate(type = factor(
    type,
    levels = c(
      "Primarily Undergraduate Institution",
      "Graduate Research Institution",
      "Non-MSI",
      "MSI"
    )
  )) %>%
  mutate(
    name = case_when(
      name == "comp_access" ~ "Access to computers and/or\ncomputational time",
      name == "course_freedom" ~ "Freedom and/or flexibility to implement a\ncourse or module of your choosing",
      name == "dev_opportunities" ~ "Opportunities for development and/or\naccess to a community of practice",
      name == "lab_access" ~ "Access to laboratory supplies to\nincorporate wet lab activities",
      name == "materials_access" ~ "Access to existing course materials",
      name == "self_confidence" ~ "Confidence and/or expertise for\nteaching the material",
      name == "student_interest" ~ "Student interest and/or engagement",
      name == "time_financial" ~ "Time and/or financial support needed to\ndesign and/or implement courses"
    )
  ) %>%
  ggplot(aes(
    x = forcats::fct_reorder(name, mean, .desc = T),
    y = mean,
    fill = type,
    group = type
  )) +
  geom_col(position = position_dodge(width = 0.9)) +
  theme(axis.text.x = element_text(angle = 30)) +
  geom_errorbar(position = position_dodge(width = 0.9),
                aes(
                  ymin = mean - sd,
                  ymax = mean + sd,
                  width = 0.4
                )) +
  labs(x = NULL, y = "Mean response") +
  scale_fill_manual(values = c("#21918c", "#3b528b", "#fde725", "#5ec962")) +
  theme_minimal() +
  theme(
    legend.title = element_blank(),
    axis.text.x = element_text(angle = 50, hjust = 1),
    plot.margin = unit(c(1, 1, 1, 2), "cm"),
    text = element_text(size = 7)
  )
gg_out

ggsave(
  "figures/fig3.png",
  height = 3,
  width = 7,
  units = "in"
)
```

# System Settings

```{r}
sessionInfo()
```

